File: src/main/java/com/graqr/threshr/model/redsky/product/PackageDimensions.java
package com.graqr.threshr.model.redsky.product;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.micronaut.serde.annotation.Serdeable;

@Serdeable
public record PackageDimensions(
        Double depth, Double height, Double weight, Double width,
        @JsonProperty("dimension_unit_of_measure")
        String dimensionUnitOfMeasure,
        @JsonProperty("weight_unit_of_measure")
        String weightUnitOfMeasure) {
}

File: src/main/kotlin/com/graqr/threshr/circleStore/model/product/PackageDimensions.kt
package com.graqr.threshr.circleStore.model.product

import com.fasterxml.jackson.annotation.JsonProperty
import io.micronaut.serde.annotation.Serdeable

@Serdeable
data class PackageDimensions(
    val depth: Double,
    val height: Double,
    val weight: Double,
    val width: Double,
    @param:JsonProperty("dimension_unit_of_measure") val dimensionUnitOfMeasure: String,
    @param:JsonProperty("weight_unit_of_measure") val weightUnitOfMeasure: String
) {}
